src/main/java/app/controllers/WarehouseController.java

```diff
@@ -0,0 +1,44 @@
+package app.controllers;
+
+import app.entities.Material;
+import app.exceptions.DatabaseException;
+import app.persistence.ConnectionPool;
+import app.persistence.MaterialMapper;
+import io.javalin.Javalin;
+import io.javalin.http.Context;
+
+import java.util.List;
+
+public class WarehouseController {
+
+    public static void addRoutes(Javalin app, ConnectionPool connectionPool) {
+        app.get("/warehouse", ctx -> viewWarehouse(ctx, connectionPool));
+        app.get("/registerNewMaterial", ctx -> ctx.render("registerNewMaterial.html"));
+        app.post("/addNewMaterial", ctx -> addNewMaterial(ctx, connectionPool));
+    }
+
+    public static void viewWarehouse(Context ctx, ConnectionPool connectionPool) {
+
+        List<Material> materialList = MaterialMapper.getAllMaterials(connectionPool);
+        ctx.attribute("materialList", materialList);
+        ctx.render("warehouse.html");
+    }
+
+    public static void addNewMaterial(Context ctx, ConnectionPool connectionPool) {
+        String description = ctx.formParam("beskrivelse");
+        String unit = ctx.formParam("enhed");
+        String materials = ctx.formParam("materiale");
+        int length = Integer.parseInt(ctx.formParam("længde"));
+        int price = Integer.parseInt(ctx.formParam("pris"));
```

Comment:
Maybe consider using english terms for the variables to keep consistency? But this is not important right now